piet-direct2d/src/d2d.rs

```diff
@@ -109,6 +120,10 @@ pub struct Effect(ComPtr<ID2D1Effect>);
 // piet-common direct2d_back, but the use cases are somewhat different.
 pub struct BitmapRenderTarget(ComPtr<ID2D1BitmapRenderTarget>);
 
+// A wrapper that exists solely to implement Hash and PartialEq over
```

Comment:
This might go easier if you add the method to D2DRenderContext (instead of DeviceContext) and use 32 bit RGBA values (instead of floats) as the basis for the cache key.